pkg/util/deployment/deployment.go

```diff
@@ -140,11 +140,21 @@ func ListPods(deployment *extensions.Deployment, getPodList podListFunc) (*api.P
 	return getPodList(namespace, options)
 }
 
+// equalIgnoreHash returns true if two given podTemplateSpec are equal, ignoring the pod-template-hash label
+// We ignore pod-template-hash because it'd be different whenever podTemplateSpec API changes
+func equalIgnoreHash(rsTemplate, template api.PodTemplateSpec) bool {
+	hash := rsTemplate.Labels[extensions.DefaultDeploymentUniqueLabelKey]
```

Comment:
Is the hash the only thing that is different because the defaulted fields get defaulted in both versions?  What other version skew possibilities are there?